iina/FileGroup.swift

```diff
@@ -11,21 +11,31 @@ import Foundation
 fileprivate let GroupPrefixMinLength = 7
 fileprivate let GroupFilenameMaxLength = 12
 
+fileprivate let charSetGroups: [CharacterSet] = [.decimalDigits, .letters]
+
 
 class FileInfo: Hashable {
   var url: URL
   var path: String
   var filename: String
   var ext: String
+  var nameInSeries: String?
   var characters: [Character]
   var dist: [FileInfo: UInt] = [:]
   var minDist: [FileInfo] = []
-  var segments: [String] = []
+  var relatedSubs: [FileInfo] = []
   var priorityStringOccurances = 0
+  var isMatched = false
 
-  var prefix: String {  // prefix detected
+  var prefix: String {  // prefix detected by FileGroup
     didSet {
-      suffix = filename.substring(from: filename.index(filename.startIndex, offsetBy: prefix.characters.count))
+      if prefix.characters.count < self.characters.count {
+        suffix = filename.substring(from: filename.index(filename.startIndex, offsetBy: prefix.characters.count))
+        getNameInSeries()
+      } else {
```

Comment:
A `prefix` longer than the file's actual name is a programmer error, right? I'd add an assertion here as well.